core/tests/test_apps.management.recovery_device.py

```diff
@@ -25,13 +25,120 @@
 
 
 class TestSlip39(unittest.TestCase):
+
     @mock_storage
-    def test_process_slip39(self):
+    def test_process_slip39_basic(self):
         storage.recovery.set_in_progress(True)
-        words = MNEMONIC_SLIP39_BASIC_20_3of6[0]
+
+        # first share (member index 5)
+        first = MNEMONIC_SLIP39_BASIC_20_3of6[0]
+        secret, share = process_slip39(first)
+        self.assertIsNone(secret)
+        self.assertEqual(share.group_count, storage.recovery.get_slip39_group_count())
+        self.assertEqual(share.iteration_exponent, storage.recovery.get_slip39_iteration_exponent())
+        self.assertEqual(share.identifier, storage.recovery.get_slip39_identifier())
+        self.assertEqual(storage.recovery.get_slip39_remaining_shares(0), 2)
+        self.assertEqual(storage.recovery_shares.get(share.index, share.group_index), first)
+
+        # second share (member index 0)
+        second = MNEMONIC_SLIP39_BASIC_20_3of6[1]
+        secret, share = process_slip39(second)
+        self.assertIsNone(secret)
+        self.assertEqual(storage.recovery.get_slip39_remaining_shares(0), 1)
+        self.assertEqual(storage.recovery_shares.get(share.index, share.group_index), second)
+        self.assertEqual(storage.recovery_shares.fetch_group(share.group_index), [second, first])  # ordered by index
+
+        # third share (member index 3)
+        third = MNEMONIC_SLIP39_BASIC_20_3of6[2]
+        secret, share = process_slip39(third)
+        self.assertIsNotNone(secret)
```

Comment:
let's check the secret is equal to the expected one

Reply:
I add a fixture, okay? Or should I call slip39.combine_mnemonics?

Reply:
yes, a constant value is OK

Reply:
Added.